File: examples/payment-service-kotlin/Payment.kt
package com.example.payment

data class Payment(
    val id: String,
    val amount: Double,
    val currency: String,
    val customerId: String,
)

sealed class PaymentError(val code: String, val message: String)
class InsufficientFundsError(message: String) : PaymentError("insufficient_funds", message)
class CardDeclinedError(message: String) : PaymentError("card_declined", message)
class TimeoutError(message: String) : PaymentError("timeout", message)
class NetworkError(message: String) : PaymentError("network", message)

File: examples/payment-service-csharp/Payment.cs
namespace PaymentService;

public record Payment(string Id, decimal Amount, string Currency, string CustomerId);

public abstract record PaymentError(string Code, string Message);
public record InsufficientFundsError(string Message) : PaymentError("insufficient_funds", Message);
public record CardDeclinedError(string Message) : PaymentError("card_declined", Message);
public record TimeoutError(string Message) : PaymentError("timeout", Message);
public record NetworkError(string Message) : PaymentError("network", Message);
